src/smk/viewer-leaflet/viewer-leaflet.js

```diff
@@ -177,47 +180,46 @@ include.module( 'viewer-leaflet', [ 'viewer', 'leaflet', 'layer-leaflet', /*'fea
     ViewerLeaflet.prototype.basemap.Streets.create = createBasemapEsri
 
     ViewerLeaflet.prototype.basemap.Imagery.create = createBasemapEsri
-    ViewerLeaflet.prototype.basemap.Imagery.labels = [ 'ImageryTransportation', 'ImageryLabels' ]
 
     ViewerLeaflet.prototype.basemap.Oceans.create = createBasemapEsri
-    ViewerLeaflet.prototype.basemap.Oceans.labels = [ 'OceansLabels' ]
-
-    ViewerLeaflet.prototype.basemap.NationalGeographic.create = createBasemapEsri
 
     ViewerLeaflet.prototype.basemap.ShadedRelief.create = createBasemapEsri
-    ViewerLeaflet.prototype.basemap.ShadedRelief.labels = [ 'ShadedReliefLabels' ]
 
     ViewerLeaflet.prototype.basemap.DarkGray.create = createBasemapEsri
-    ViewerLeaflet.prototype.basemap.DarkGray.labels = [ 'DarkGrayLabels' ]
 
     ViewerLeaflet.prototype.basemap.Gray.create = createBasemapEsri
-    ViewerLeaflet.prototype.basemap.Gray.labels = [ 'GrayLabels' ]
-
+    
     ViewerLeaflet.prototype.basemap.StamenTonerLight.create = createBasemapTiled
     ViewerLeaflet.prototype.basemap.StamenTonerLight.url = 'https://stamen-tiles-{s}.a.ssl.fastly.net/toner-lite/{z}/{x}/{y}.png'
     ViewerLeaflet.prototype.basemap.StamenTonerLight.attribution = "Map tiles by <a href='http://stamen.com'>Stamen Design</a>, under <a href='http://creativecommons.org/licenses/by/3.0'>CC BY 3.0</a>. Data by <a href='http://openstreetmap.org'>OpenStreetMap</a>, under <a href='http://www.openstreetmap.org/copyright'>ODbL</a>."
 
-    function createBasemapEsri( id ) {
+    function createBasemapEsri( esriApiKey, apiId ) {
+        if (!esriApiKey) {
+            throw new Error('No value was found for "esriApiKey" while creating an ESRI vector basemap. As a result, no basemap will appear. Please confirm there is a value for "esriApiKey" in the "viewer" section of smk-config.json, or use a non-ESRI basemap.');
+        }
+
         /* jshint -W040 */
-        var opt = Object.assign( { detectRetina: true }, this.option )
+        var opt = Object.assign( { detectRetina: true }, this.option );
+        opt.pane = 'basemaps';
```

Comment:
Rather than using a hard-coded string here (and above at lines 32 and 33 and below at line 219), consider defining a constant that can be used at each spot.

Reply:
Done. I define the const high in the class (outside of where it's first used in ViewerLeaflet.prototype.initialize) so that it is visible in other functions.